src/hazelcore/semantics/ErrStatus.re

```diff
@@ -1,9 +1,12 @@
+open Sexplib.Std;
+
 module HoleReason = {
   /* Variable: `reason` */
   [@deriving sexp]
   type t =
     | TypeInconsistent
-    | WrongLength;
+    | WrongLength
+    | InconsistentBranches(list(HTyp.t));
```

Comment:
it's cleaner to have a separate CaseErrStatus.t that only appears on case than to have InconsistentBranches show up in all the pattern matches. TypeInconsistent can appear on any form, and WrongLength can appear on any opseq containing commas, but InconsistentBranches can only appear on one form.

Reply:
Would this be similar to what is going on with VarErrStatus? (in the sense that there is an error status that only appears on a particular form)

Reply:
yeah though I think it would be better to replace the ErrStatus.t on Case with a CaseErrStatus.t that has one constructor for standard errors and then one more for the new one.

Reply:
I am not exactly sure how to just replace ErrStatus on Case. It seems like there are places where it is expected that all of the forms will have an ErrStatus on them... Maybe I am just misunderstanding what you mean. 
Why was the VarErrStatus done the way it was if it is better to just replace ErrStatus?

Reply:
Yeah you'd have to look at all the places where there is a get_err_status call and handle the InconsistentBranches situation separately, if it is relevant there. IIRC that function isn't used much and when it is, it is on opseqs, but I might be wrong. If it's too much work to refactor then don't worry about it.

VarErrStatus should probably be done this way too.

Reply:
This commit has my attempt at these changes. https://github.com/hazelgrove/hazel/pull/215/commits/f15a75ac87e3dc13f282dde1a1f8520785571066